### PR TITLE
Add field/line delimiter property in create hive table

### PR DESCRIPTION
**Pull request by yuananf**
This is useful when creating a TEXTFILE table.

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveTableProperties.java [hunk 1]
```diff
@@ -39,6 +41,8 @@
     public static final String PARTITIONED_BY_PROPERTY = "partitioned_by";
     public static final String BUCKETED_BY_PROPERTY = "bucketed_by";
     public static final String BUCKET_COUNT_PROPERTY = "bucket_count";
+    public static final String FIELD_DELIM_PROPERTY = "field_delim";
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveTableProperties.java]: I'm thinking we could call these `fields_delimited_by` and `lines_terminated_by`

Comment (martint) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveTableProperties.java]: How about field_delimiter and line_delimiter?

Comment (yuananf) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveTableProperties.java]: I prefer ```field_delimiter``` and ```line_delimiter```.

Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveTableProperties.java]: Sounds good to me.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveTableProperties.java [hunk 2]
```diff
@@ -39,6 +41,8 @@
     public static final String PARTITIONED_BY_PROPERTY = "partitioned_by";
     public static final String BUCKETED_BY_PROPERTY = "bucketed_by";
     public static final String BUCKET_COUNT_PROPERTY = "bucket_count";
+    public static final String FIELD_DELIM_PROPERTY = "field_delimiter";
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveTableProperties.java]: Don't abbreviate variable names. This should be `FIELD_DELIMITER_PROPERTY`.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveTableProperties.java [hunk 3]
```diff
@@ -77,7 +81,9 @@ public HiveTableProperties(TypeManager typeManager, HiveClientConfig config)
                                 .map(name -> ((String) name).toLowerCase(ENGLISH))
                                 .collect(Collectors.toList())),
                         value -> value),
-                integerSessionProperty(BUCKET_COUNT_PROPERTY, "Number of buckets", 0, false));
+                integerSessionProperty(BUCKET_COUNT_PROPERTY, "Number of buckets", 0, false),
+                stringSessionProperty(FIELD_DELIM_PROPERTY, "Field delimiter", "\001", false),
+                stringSessionProperty(LINE_DELIM_PROPERTY, "Line delimiter", "\002", false));
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveTableProperties.java]: The default line delimiter should be newline (`\n`)

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveTableProperties.java [hunk 4]
```diff
@@ -117,4 +123,14 @@ public static HiveStorageFormat getHiveStorageFormat(Map<String, Object> tablePr
     {
         return (List<String>) tableProperties.get(BUCKETED_BY_PROPERTY);
     }
+
+    public static String getFieldDelimProperty(Map<String, Object> tableProperties)
+    {
+        return StringEscapeUtils.unescapeJava((String) tableProperties.get(FIELD_DELIM_PROPERTY));
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveTableProperties.java]: We shouldn't have a special escape format for strings. Users can specify this in SQL:
```sql
CREATE TABLE ... WITH (field_delimiter = chr(1))
```
For printing in the SQL formatter, we should should detect when a string literal contains non-printable characters and switch to the escaped Unicode format (which needs to be implemented in Presto): https://github.com/prestodb/presto/issues/6585

**Comment by electrum** [comment]
This should only be allowed for text file and fail otherwise.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveTableProperties.java (hunk 4 above)
Comment (yuananf) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveTableProperties.java]: Sometimes we create table with field delimiter '\t', the field delimiter of target table will be '\\t' without unescape. Is there a better way to handle this?

**Comment by electrum** [comment]
You can use chr(9) since properties are SQL expressions. We should also raise an error if the delimiter is not exactly one character.

**Comment by col15** [comment]
Is it possible to also allow specification of compression type?

**Comment by drummerwolli** [comment]
any chance of this being done at some point? or is there an alternative by now to achieve the same goal?

**Comment by d3cay1** [comment]
This is a useful feature for my team.  Why is this taking so long to incorporate (besides now having conflicts)?

**Comment by kgalieva** [comment]
@yuananf thank you for this commit, this is a really useful feature! Do you have any plans to fix test failures and resolve conflicts? If you don't have time, please let me know, I will create another PR with necessary changes, because we need an opportunity to create tables with custom delimiter. Thank you!

**Comment by raghavsethi** [comment]
Closing since it appears @kgalieva will open another PR to supersede this.

**Comment by nguyenminhtuanfit** [comment]
@kgalieva , @raghavsethi any update on this topic

**Comment by raghavsethi** [comment]
Doesn't look like a new PR was submitted. Feel free to create one.

**Comment by WingsLong** [comment]
why this change do not merge to the master

**Pull request closed without merging**